lib/IRGen/GenHasSymbol.cpp

```diff
@@ -74,6 +88,17 @@ class HasSymbolIRGenVisitor : public IRSymbolVisitor {
   }
 
   void addLinkEntity(LinkEntity entity) override {
+    if (entity.hasSILFunction()) {
+      addFunction(entity.getSILFunction());
+      return;
+    }
+
+    // FIXME: These function impl link entities for @_dynamicReplacement(for:)
+    // functions don't appear to be used outside of TBDGen. Ignore them for
+    // the purposes of #_hasSymbol.
+    if (entity.isDynamicallyReplaceableFunctionImpl())
```

Comment:
@aschwaighofer I could use your help understanding why I don't see `LinkEntity::forDynamicallyReplaceableFunctionImpl()` used anywhere in the compiler besides `IRSymbolVisitorImpl`, which just influences TBD generation and `#_hasSymbol` code generation right now.

Reply:
For `SILFunction`s rather than AST decls the mangling is handled as part of the `Linking::Kind::SILFunction` kind.

```
  case Kind::SILFunction: {                                                     
    std::string Result(getSILFunction()->getName());                            
    if (isDynamicallyReplaceable()) {                                           
      Result.append("TI");                                                      
    }                                                                           
    return Result;                                                              
  }  
  ```